### PR TITLE
Extracted sync from upload_qpysequence into an independent line

### PR DESCRIPTION
**Pull request by jordivallsq**
Moved the sync outside of upload_qpysequence in Qblox_module into an independent function sync_sequencers located in the same .py file mirroring the function desync_sequencers and added 3 lines in platform.py to also mirror the desync and sync every time the execution is ran.

### CODE REVIEW

**Comment by jordivallsq** [comment]
The code has been checked with 2 different pulses from different sequencers in a software loop.
This is the n step of a software loop where we can see how both pulses are in sync:
imagen

The sequencers used are:
`                wait_sync        4              

main:
                move             200, R0        
avg_0:
                wait             2000           
                wait             4              
                play             0, 1, 4        
                acquire_weighed  0, 0, 0, 0, 2000
                wait             200            
                loop             R0, @avg_0     
                stop                            


[qililab] [0.23.3|INFO|2024-03-06 10:18:25]: Sequence program: 
 setup:
                wait_sync        4              

main:
                move             200, R0        
avg_0:
                play             0, 1, 2000     
                wait             2208           
                loop             R0, @avg_0     
                stop     `

We can see how the sequencers structure follow the pulses distribution

Using Qililab main we are not able to see one of the colors in the image, only one is seen because they desync

File: src/qililab/platform/platform.py [hunk 1]
```diff
@@ -652,6 +652,11 @@ def _execute_qprogram_with_qblox(
                 buses[bus_alias].upload_qpysequence(qpysequence=sequences[bus_alias])
                 self._qpy_sequence_cache[bus_alias] = sequence_hash
 
+        # sync all instruments
+        for instrument in self.instruments.elements:
+            if isinstance(instrument, QbloxModule):
+                instrument.sync_sequencers()
```
Comment (fedonman) [inline src/qililab/platform/platform.py:658]: If you do this, you will enable syncing to **all** sequencers present. We only want to enable syncing to the sequencers that are relevant to the specific QProgram.

The de-sync is happening for all sequencers, since you don't really care about it (it doesn't affect anyone if you desync an irrelevant sequencer), but you may change that too so it's more consistent.

Comment (jordivallsq) [reply src/qililab/platform/platform.py]: Now I set the sync as true inside the bus loop (`for bus_alias in sequences:`) where the `upload_qpysequence` is located (it is not inside the conditional) with:
```
for instrument in buses[bus_alias].system_control.instruments:
    if isinstance(instrument, QbloxModule):
        instrument.sync_sequencers(buses[bus_alias].port)
```
All the required information (port and instruments) comes from the bus

And now the desync is also like this:
```
# Reset instrument settings
for bus_alias in sequences:
    for instrument in buses[bus_alias].system_control.instruments:
        if isinstance(instrument, QbloxModule):
            instrument.desync_sequencers(buses[bus_alias].port)
```

The functions changed inside QbloxModule (since these are only rellevant here) are:
```
def sync_sequencers(self, port: str) -> None:
    """Syncs all sequencers."""
    sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
    for sequencer in sequencers:
        self.device.sequencers[sequencer.identifier].sync_en(True)

def desync_sequencers(self, port: str) -> None:
    """Syncs all sequencers."""
    sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
    for sequencer in sequencers:
        self.device.sequencers[sequencer.identifier].sync_en(False)
```

Comment (jordivallsq) [reply src/qililab/platform/platform.py]: I have also tested the syncing

imagen

but the code saving still gives me errors by changing the imports

File: src/qililab/instruments/qblox/qblox_module.py [hunk 2]
```diff
@@ -70,9 +70,9 @@ def __post_init__(self):
                 )
 
             self.awg_sequencers = [
-                AWGQbloxSequencer(**sequencer)
-                if isinstance(sequencer, dict)
-                else sequencer  # pylint: disable=not-a-mapping
+                (
+                    AWGQbloxSequencer(**sequencer) if isinstance(sequencer, dict) else sequencer
+                )  # pylint: disable=not-a-mapping
```
Comment (fedonman) [inline src/qililab/instruments/qblox/qblox_module.py:75]: why is this changed? I think it was messed by your formatter before you set it up correctly. Could you revert it back?

File: src/qililab/instruments/qblox/qblox_module.py [hunk 3]
```diff
@@ -111,6 +111,19 @@ def initial_setup(self):
         for idx, offset in enumerate(self.out_offsets):
             self._set_out_offset(output=idx, value=offset)
 
+    def sync_by_port(self, port: str) -> None:
+        """Syncs all sequencers."""
+        sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
+        for sequencer in sequencers:
+            self.device.sequencers[sequencer.identifier].sync_en(True)
+
+    # TODO: merge both desyncs or decide if desync_sequencers is enough
```
Comment (fedonman) [inline src/qililab/instruments/qblox/qblox_module.py:120]: comment is not necessary here since we keep and use both methods.

File: tests/.pylintrc [hunk 4]
```diff
@@ -19,4 +19,5 @@ disable=
     too-many-public-methods,
     missing-class-docstring,
     missing-function-docstring,
-    import-error
+    import-error,
+    R0912
```
Comment (fedonman) [inline tests/.pylintrc:23]: add the rule by its name, not its id.

File: .pylintrc [hunk 5]
```diff
@@ -19,4 +19,5 @@ disable=
     duplicate-code,
     too-many-ancestors,
     missing-module-docstring,
-    too-many-arguments
+    too-many-arguments,
+    R0912
```
Comment (fedonman) [inline .pylintrc:23]: add the rule by its name, not its code

**Review by fedonman — CHANGES_REQUESTED** [review]
Awesome job! 💯 Requested some minor changes

**Review by fedonman — APPROVED** [review]
Awesome! 💯
